### ISSUE TITLE
MNN的embedding操作是否支持

### ISSUE THREAD

**Issue by yananchen1989** [alibaba/MNN#1543]
Hello，我在MNN文档里尚未找到embedding的相关操作，请问未来是否有计划支持这一功能？输入整数id，对其进行embedding lookup的操作使其transform到实数embedding向量？

**Comment by jxt1234** [comment]
embedding 对应 gather / onehot 等算子，是支持的

**Comment by yananchen1989** [comment]
> embedding 对应 gather / onehot 等算子，是支持的

你好，我的目标是在device上训练，也包括训练embedding。
比如对于文本分类,输入的是对齐的word-id，整数，希望能在内置的embedding matrix中lookup，同时训练网络的同时，更新embedding。

> 87,12,13,24,10 business
> 77,34,12,98,4 technology
> 34,21,28,76,0 politics

**Comment by jxt1234** [comment]
这个之前有需求，就是 Gather

**Comment by jxt1234** [comment]
用 GatherV2 的话支持的

**Comment by yananchen1989** [comment]
> 用 GatherV2 的话支持的

可否提供一个简明的例子作为参考呢？

**Comment by jxt1234** [comment]
你是需要用 MNN 构图还是转模型?

**Comment by yananchen1989** [comment]
> 你是需要用 MNN 构图还是转模型?

直接用MNN构图。我看到gather可以根据index去取对应的embedding（假设embedding是预先设定的参数矩阵，N*K大小，N为词表大小，K为embedding尺寸，比如64）。
我想问的是，在gather之后，这些对应的embedding参数是否会随着下游模型一起训练、梯度更新。

**Comment by jxt1234** [comment]
目前只支持少量情况（GatherV2 且 axis = 0），后面会重构训练求导部分彻底支持

**Comment by T122D002L** [comment]
> 目前只支持少量情况（GatherV2 且 axis = 0），后面会重构训练求导部分彻底支持

请问这个部分目前最新的进展如何呢？我在转换将torch..nn.Embedding 转换的mnn模型到端侧运行时，模型的输出使用readMap读取的结果是空指针
